vulnerabilities/importers/apache_httpd.py

```diff
@@ -49,47 +53,58 @@ def updated_advisories(self):
+                advisories.append(self.to_advisory(data))
             return self.batch_advisories(advisories)
 
         return []
 
-
-def to_advisories(data):
-    advisories = []
-    for issue in data:
+    def to_advisory(self, data):
+        cve = data["CVE_data_meta"]["ID"]
+        description = data["description"]["description_data"]
+        summary = next((item["value"] for item in description if item["lang"] == "eng"), "")
+        severity = VulnerabilitySeverity(
+            system=scoring_systems["apache_httpd"],
+            value=data["impact"][0]["other"],
+        )
+        reference = Reference(reference_id=cve, url=self.url + cve + ".json", severities=[severity])
         resolved_packages = []
         impacted_packages = []
-        for info in issue:
-            if info.tag == "cve":
-                cve = info.attrib["name"]
-
-            if info.tag == "title":
-                summary = info.text
-
-            if info.tag == "fixed":
-                resolved_packages.append(
-                    PackageURL(type="apache", name="httpd", version=info.attrib["version"])
-                )
-
-            if info.tag == "affects" or info.tag == "maybeaffects":
-                impacted_packages.append(
-                    PackageURL(type="apache", name="httpd", version=info.attrib["version"])
-                )
-
-        advisories.append(
-            Advisory(
-                vulnerability_id=cve,
-                summary=summary,
-                impacted_package_urls=impacted_packages,
-                resolved_package_urls=resolved_packages,
-            )
-        )
 
-    return advisories
+        for vendor in data["affects"]["vendor"]["vendor_data"]:
+            for products in vendor["product"]["product_data"]:
+                for version in products["version"]["version_data"]:
+                    version_value = version["version_value"]
+
+                    if version["version_affected"] == "<":
+                        resolved_packages.append(
+                            PackageURL(type="apache", name="httpd", version=version_value)
+                        )
+
+                    else:
+                        impacted_packages.append(
+                            PackageURL(type="apache", name="httpd", version=version_value)
+                        )
+
+        return Advisory(
+            vulnerability_id=cve,
+            summary=summary,
+            impacted_package_urls=impacted_packages,
+            resolved_package_urls=resolved_packages,
+            references=[reference],
+        )
 
 
-def fetch_xml(url):
-    resp = requests.get(url).content
-    return ElementTree.fromstring(resp)
+def fetch_links(url):
+    links = []
+    data = requests.get(url).content
+    soup = BeautifulSoup(data, features="lxml")
+    for tag in soup.find_all("a"):
+        link = tag.get("href")
+        if not re.search("^CVE.*json$", link):
+            continue
+        links.append(url + link)
```

Comment:
A robust option would be to use 
https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urljoin instead of  string concat.